### ISSUE TITLE
When converting SArray of strings to float, strings starting with inf will become float('inf')

### ISSUE THREAD

**Issue by guy4261** [apple/turicreate#352]
Hello,

If your `str`-typed SArray contains only strings starting with `'inf'` and you are trying to convert it to float, what you will get instead of a raised exception is an array of the 'infinite' float.

This happened to me while I was going through my columns and using try/catch to see which ones I could convert to float. Not a very common use case, I admit, but still...

```python
$ ipython
Python 2.7.10 (default, Jul 15 2017, 17:16:57)
Type "copyright", "credits" or "license" for more information.

IPython 5.5.0 -- An enhanced Interactive Python.
?         -> Introduction and overview of IPython's features.
%quickref -> Quick reference.
help      -> Python's own help system.
object?   -> Details about 'object', use 'object??' for extra details.

In [1]: import turicreate as tc

In [2]: tc.__version__
Out[2]: '4.2'

In [3]: float('inf')  # the expected result
Out[3]: inf

In [4]: tc.SArray(['inf', 'infamous', 'infrequent', 'INFINITE']).astype(float)  # some actual mishaps beside it
Out[4]:
dtype: float
Rows: 4
[inf, inf, inf, inf]

In [5]: tc.SArray(['foo']).astype(float) # what I expect
...traceback omitted...
RuntimeError: Unable to interpret foo as the target type.
```

**Comment by ylow** [comment]
This is a side effect of an unfortunate simple direct use of strtod.

More generally, this is occurs for strings prefixed with numbers as well
```
In [5]: tc.SArray(["2hello","2.3world"]).astype(float)
Out[5]:
dtype: float
Rows: 2
[2.0, 2.3]

In [6]: tc.SArray(["2hello","2.3world"]).astype(int)
Out[6]:
dtype: int
Rows: 2
[2, 2]
```

**Comment by shantanuchhabra** [comment]
@guy4261 Thanks for pointing this out! Fixed by PR #373
